### PR TITLE
Improving entities contracts

### PR DESCRIPTION
**Pull request by yceruto**
For example:
```php
$post = new Post();

// ...

if (!$post->getAuthor()) { // it's null initially so it should be fine allow null return
    // ...
} 

// ...

$post->setAuthor(null);
// shouldn't be allowed since the column is not null -> on flush() it'll fail anyway
```
Also it's consistent with `make:entity` command.

### CODE REVIEW

File: src/Entity/Comment.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ public function getPost(): ?Post
         return $this->post;
     }
 
-    public function setPost(?Post $post): void
+    public function setPost(Post $post): void
```
Comment (stof) [inline src/Entity/Comment.php]: that breaks `Post::removeComment` (but the proper way there is not to set it as `null` here anyway, as the Comment will be removed entirely by the `orphanRemoval` feature, not be kept orphan)

Comment (yceruto) [reply src/Entity/Comment.php]: Thanks @stof, done.

**Comment by yceruto** [comment]
@javiereguiluz what do you think about this proposal?

File: src/Entity/Comment.php [hunk 2]
```diff
@@ -105,7 +105,7 @@ public function setContent(string $content): void
         $this->content = $content;
     }
 
-    public function getPublishedAt(): \DateTime
+    public function getPublishedAt(): ?\DateTime
```
Comment (stof) [inline src/Entity/Comment.php]: not needed, as the field is not nullable in the DB and the constructor initializes it

File: src/Entity/Post.php [hunk 3]
```diff
@@ -158,27 +158,27 @@ public function getContent(): ?string
...
     }
 
-    public function setContent(?string $content): void
+    public function setContent(string $content): void
     {
         $this->content = $content;
     }
 
-    public function getPublishedAt(): \DateTime
+    public function getPublishedAt(): ?\DateTime
```
Comment (stof) [inline src/Entity/Post.php]: same here

**Comment by yceruto** [comment]
@stof comments addressed, thanks.

File: src/Entity/Post.php [hunk 4]
```diff
@@ -198,7 +198,6 @@ public function addComment(?Comment $comment): void
 
     public function removeComment(Comment $comment): void
     {
-        $comment->setPost(null);
```
Comment (voronkovich) [inline src/Entity/Post.php]: @yceruto, Why did you remove this line?

Comment (yceruto) [reply src/Entity/Post.php]: See https://github.com/symfony/demo/pull/899#discussion_r235046319

Comment (voronkovich) [reply src/Entity/Post.php]: @yceruto, Ok, I got it.

**Comment by javiereguiluz** [comment]
Thank you Yonel.
